File: Data Structures & Algorithms/reverse-bits/submission-0.java
class Solution {
    public int reverseBits(int n) {
        int res = 0;
        for(int i = 0 ; i < 32; i++){
            int bit = (n >> i) & 1;
            bit <<= (31 - i);
            res += bit;
        }
        return res;
    }
}

File: Data Structures & Algorithms/combinations/submission-0.py
class Solution:
    def combine(self, n: int, k: int) -> List[List[int]]:
        res = []

        def helper(i, subset):
            if len(subset) == k:
                res.append(subset.copy())
                return
            if i > n:
                return
            
            for j in range(i, n+1):
                subset.append(j)
                helper(j+1, subset)
                subset.pop()
        
        helper(1, [])
        return res
            